### ISSUE TITLE
No pathfinding, wants to walk through walls

### ISSUE THREAD

**Issue by nhitzert69** [ldtteam/minecolonies#10477]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.20

### MineColonies Version

1.1.747

### Structurize Version

1.0.760

### Current Behavior

All of my workers constantly walk into walls, for example my cook wants to go inside vut instead walks into a the wall, then after 5 seconds of jumping he walks into another wall and keeps doing this untill after maybe 2 minutes he walks inside. Non of my npcs sit down in the chairs of the restaurant, they also walk into the restaurant wall and keep jumping untill they’re served food. My builders do do their job, my farmer, woodgetter and miner do do their jobs but they also walk somewhere with no trees or into a wall insted of to the farm, however they do still do their job eventually. My courriers however walk towards a wall instead of into a place, or when i ask them to come they walk towards the place but instead of following the path they walk into a building then wait a couole seconds before going back to the warehouse. My entire town is surrounded with walls so no monsters can get in, this also means that from one part to another of the town ive made a special path which they can follow and its the only path. Npcs also keep walking into lanterns and then jumping up and down instead of walking past, so far i have a couple hours in the game and the workers all have stuff in their storage units. But the warehouse is completely empty, so i think they are unable to go to places. Is there maybe a way for me to mark a path so the npcs can only walk on those blocks?

### Expected Behavior

The expected behavior is that they dont walk into walls and just follow the ground

### Reproduction Steps

(update: I just tried creating a new world, placed the colonial town hall restaurant and warehouse and requested pickup at the restaurant to try and it did the exact same issue, so you dont have to create those entire towns as I put in the steps)
0. I used to have minecolonies tweaks installed, but removed it after maybe 10 minutes of playing
1.i use colonial
 I have one town with all residential buildings and warehouse, university etc. In this town almost all buildings are connected to one another leaving no space in between. All other blocks are paths and where there are no buildings i’ve put walls to completely block any exit. Then i have a town a little further away where the miner, forester, farmer and their residential building is located. These buildings are not connected and theres a lot of space in between, here the building dont block off any exit so the entire town is surrounded by walls.
2. Both towns ar completly surrounded with walls so no one can get outside the walls, town one alot of buildings act as walls to block off any potential exit, where there are no buildings there are walls. Town two only walls.
3. Then i’ve made a path from town one to town two, the path is surrounded by leaves as walls and cealing 

### Logs

https://gist.github.com/nhitzert69/09e156026affdb7bc13675d5ff51ee1c

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by RewireYourBrain** [comment]
Ive been having this issue as well. Its saying that you marked it as completed, how did you end up fixing it?

**Comment by nhitzert69** [comment]
> Ive been having this issue as well. Its saying that you marked it as completed, how did you end up fixing it?

I marked it as complete as the full story was a bit messy. The restaurant is still a bit weird qs they’ll still walk into walls. Turns out I had too many walls and too little place for them to get from a to b. 

I removed the wall with the path and instead made a bigger wall around my city.

since my buildings in town 1 are so close together and they only had 1 way to go to town 2 they couldnt find the path.

Now that they have more space they have no trouble moving around and they can find their way better.

My npcs still walk into walls and walk into objects on the street. (Mainly fences and walls blocks). But if i notice it happens too often I just move them. Or they sometimes try to climb up a two block high instead of going around it so if i see it happen too often i just change it to make it easier.

Usually after a while of tryin to go through a wall eventually they go around it (can sometimes take a couple minutes which is kinda annoying)

So long story short they need space to move and ease of movement.

And I marked it as complete since It would be better to make a new issue with better details as my comment was a little rushed.

Hope this still makes things a bit clear for you.
